[TEP-0091] change feature flag resource-verification-mode to trusted-resources-verification-no-match-policy (#6324)
tektoncd/pipeline: docs/trusted-resources.md

    app.kubernetes.io/instance: default
    app.kubernetes.io/part-of: tekton-pipelines
data:
  trusted-resources-verification-no-match-policy: "fail"
```

`trusted-resources-verification-no-match-policy` configurations:

>  * `ignore`: Don't fail the taskrun/pipelinerun if no matching policies are found. Skip the verification and don't log.
>  * `warn`: Don't fail the taskrun/pipelinerun if no matching policies are found. Skip the verification and log a warning

Reviewer: ```suggestion
 * `ignore`: if no matching policies are found, skip the verification, don't log, and don't fail the taskrun/pipelinerun
 * `warn`:  if no matching policies are found, skip the verification, log a warning, and don't fail the taskrun/pipelinerun
```

Just trying to make it extra clear that verification is only skipped when no matching policies are found

Author: Thank you!

 * `fail`: Fail the taskrun/pipelinerun if no matching policies are found.

 **Notes:**

>  * To skip the verification: make sure if no policies exist and `trusted-resources-verification-no-match-policy` is not set to `fail`.

Reviewer: ```suggestion
 * To skip the verification: make sure no policies exist and `trusted-resources-verification-no-match-policy` is set to `warn` or `ignore`.
```

>  * To enable the verification: install VerificationPolicy to match the resources.

Author: The verificationPolicies examples is right after this section so I didn't link it. But you're right. I added a reference to the next section. 

>  do the verification policies have to be in the same namespace as the resource being verified

Yes, it is mentioned in the next section
